src/compiler/checker.ts

```diff
@@ -48739,7 +48735,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     // Keep this up-to-date with the same logic within `getApparentTypeOfContextualType`, since they should behave similarly

```

Comment:
Do these changes preserve this logic?

Reply:
Yes, I think that just speaks to the fact that they all use `discriminateTypeByDiscriminableItems`.